bookingmanagerpersistence/src/main/java/cz/fi/muni/pa165/entity/User.java

```diff
@@ -96,4 +104,48 @@ public void setRole(Role role) {
     public void setReservations(Set<Reservation> reservations) {
         this.reservations = reservations;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof User)) return false;
+
+        User user = (User) o;
+
+        if (id != user.id) return false;
+        if (!fullName.equals(user.fullName)) return false;
+        if (!email.equals(user.email)) return false;
+        if (phoneNumber != null ? !phoneNumber.equals(user.phoneNumber) : user.phoneNumber != null) return false;
+        if (address != null ? !address.equals(user.address) : user.address != null) return false;
+        if (!passwordHash.equals(user.passwordHash)) return false;
+        if (role != user.role) return false;
+        return reservations != null ? reservations.equals(user.reservations) : user.reservations == null;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Tohle by se podle Adamka melo implementovat pres `Objects.hashCode`.

Reply:
Nejsem si jistý, jestli rozumím. Máš prosím příklad?

Reply:
Něco takového:    
@Override
    public int hashCode() {
        int result = Objects.hashCode(fullName);
        result = 31 * result + Objects.hashCode(email);
        result = 31 * result + (phoneNumber != null ? Objects.hashCode(phoneNumber) : 0);
        result = 31 * result + (address != null ? Objects.hashCode(address) : 0);
        result = 31 * result + Objects.hashCode(passwordHash);
        result = 31 * result + Objects.hashCode(role);
        result = 31 * result + (reservations != null ? Objects.hashCode(reservations) : 0);
        return result;
    }

Reply:
https://www.mkyong.com/java/java-how-to-overrides-equals-and-hashcode/

2. JDK 7 
Akorát přístup k fieldum přes get metody

Reply:
Takže jenom: return Objects.hash(getFullName(), getAddress(), getEmail(), getPasswordHash(), getPhoneNumber(), getReservations(), getRole()); ?

Reply:
Jo, super